src/lib.rs

```diff
@@ -101,6 +110,17 @@ where
     let _ = circuit_secondary.synthesize(&mut cs);
     let (r1cs_shape_secondary, r1cs_gens_secondary) = (cs.r1cs_shape(), cs.r1cs_gens());
 
+    let a1 = if A1::to_usize() == 1 {
+      None
+    } else {
+      Some(PoseidonConstants::<<G1 as Group>::Scalar, A1>::new())
```

Comment:
Do we need to check if the arity is less than a certain number? Or is the call to PoseidonConstants always going to succeed?

Reply:
It won't fail at runtime. If you try to specify a concrete array size that doesn't have an implementation of Arity in Neptune (maybe > 36?) then it will fail at compile time. So I don't think a runtime check is required.